lib/security/WSSecurity.js

```diff
@@ -3,10 +3,21 @@
 var crypto = require('crypto')
   , passwordDigest = require('../utils').passwordDigest;
 
-function WSSecurity(username, password, passwordType) {
+function WSSecurity(username, password, options) {
+  options = options || {};
   this._username = username;
   this._password = password;
-  this._passwordType = passwordType || 'PasswordText';
+  //must account for backward compatibility for passwordType String param as well as object options defaults: passwordType = 'PasswordText', hasTimeStamp = true   
+  if (typeof options === 'string') {
+    this._passwordType = options;
+    this._passwordType = this._passwordType === 'PasswordDigest' ? 'PasswordDigest' : 'PasswordText';
+  } else {
+    this._passwordType = typeof options.passwordType !== 'undefined' ? options.passwordType : 'PasswordText';
```

Comment:
thanks a lot for your patience @jayharper :+1: 

I think it's completely viable to set `this._passwordType` and `this._hasTimeStamp` properties once without setting them and resetting them within the next line. So if you please -  as a last step - could change the "double set"-lines with "one-liners" like:

```javascript
if (typeof options === 'string') {
  this._passwordType = options ? options : 'PasswordText';
} else {
  this._passwordType = options.passwordType ? options.passwordType : 'PasswordText';
}

this._hasTimeStamp = !!options.hasTimeStamp; // implicit Boolean conversion!
```

This would a) shorten the code and b) make it more readable. Thanks a lot for your understanding :smiley: